crmsh/bootstrap.py

```diff
@@ -331,17 +334,36 @@ def _update_configuration(self):
         Update /etc/sysconfig/sbd
         """
         shutil.copyfile(self.SYSCONFIG_SBD_TEMPLATE, SYSCONFIG_SBD)
+        self._determine_sbd_watchdog_timeout()
         sbd_config_dict = {
                 "SBD_PACEMAKER": "yes",
                 "SBD_STARTMODE": "always",
                 "SBD_DELAY_START": "no",
                 "SBD_WATCHDOG_DEV": self._watchdog_inst.watchdog_device_name
                 }
+        if self._sbd_watchdog_timeout > 0:
+            sbd_config_dict["SBD_WATCHDOG_TIMEOUT"] = str(self._sbd_watchdog_timeout)
         if self._sbd_devices:
             sbd_config_dict["SBD_DEVICE"] = ';'.join(self._sbd_devices)
         utils.sysconfig_set(SYSCONFIG_SBD, **sbd_config_dict)
         csync2_update(SYSCONFIG_SBD)
 
+    def _determine_sbd_watchdog_timeout(self):
+        """
+        When using diskless SBD, determine value of SBD_WATCHDOG_TIMEOUT
+        """
+        if not self.diskless_sbd:
+            return
+        # add sbd after qdevice started
+        if utils.is_qdevice_configured() and utils.service_is_active("corosync-qdevice.service"):
+            qdevice_sync_timeout = utils.get_qdevice_sync_timeout()
+            self._sbd_watchdog_timeout = qdevice_sync_timeout + 5
```

Comment:
I think we need to consider s390(x) in here. I need to make sure  _sbd_watchdog_timeout >= 15s in case of s390(x).

Reply:
@gao-yan Here `qdevice_sync_timeout` by default is 30s, then `_sbd_watchdog_timeout` is 35, already >= 15s
Do you mean in s390 case, if value of `qdevice_sync_timeout` is small, like 5, we should make sure `sbd_wachdog_timeout` >= 15? Changed like this?
```Python
            qdevice_sync_timeout = utils.get_qdevice_sync_timeout()
            self._sbd_watchdog_timeout = qdevice_sync_timeout + 5
            if self._is_s390 and self._sbd_watchdog_timeout < 15:
                self._sbd_watchdog_timeout = 15
```

Reply:
Yes. IIUC, sbd stage can run for an existing cluster? That's also the reason why rather than relying on the default value, we are getting the real qdevice sync_timeout with get_qdevice_sync_timeout() just in case it was configured differently or changed somehow ...

Reply:
OK, changed